app/models/pipeline_run.rb

```diff
@@ -1197,6 +1272,11 @@ def self.download_file(s3_path, destination, dest_is_dir = true)
   end
 
   def file_generated(s3_path)
+    unless s3_path
```

Comment:
Would you mind leaving `file_generated` untouched and instead doing the necessary validations in your calling of it?